bin/alluxio-mount.sh

```diff
@@ -85,44 +75,100 @@ function mount_ramfs_linux() {
   fi
 }
 
+function umount_ramfs_linux() {
+  if mount | grep ${TIER_PATH} > /dev/null; then
+    echo "Unmounting ${TIER_PATH}"
+    if [[ ${USE_SUDO} == true ]]; then
+      sudo umount -l -f ${TIER_PATH}
+    else
+      umount -l -f ${TIER_PATH}
+    fi
+    if [[ $? -ne 0 ]]; then
+      echo "ERROR: umount RamFS ${TIER_PATH} failed" >&2
+      exit 1
+    fi
+  fi
+}
+
 function mount_ramfs_mac() {
   # Convert the memory size to number of sectors. Each sector is 512 Byte.
   local num_sectors=$(${BIN}/alluxio runClass alluxio.util.HFSUtils ${MEM_SIZE} 512)
 
   # Format the RAM FS
-  # We may have a pre-existing RAM FS which we need to throw away
   echo "Formatting RamFS: ${TIER_PATH} ${num_sectors} sectors (${MEM_SIZE})."
+  # Remove the "/Volumes/" part so we can get the name of the volume.
+  diskutil erasevolume HFS+ ${TIER_PATH/#\/Volumes\//} $(hdiutil attach -nomount ram://${num_sectors})
+}
+
+function umount_ramfs_mac() {
   local device=$(df -l | grep ${TIER_PATH} | cut -d " " -f 1)
   if [[ -n "${device}" ]]; then
+    echo "Unmounting ramfs at ${TIER_PATH}"
     hdiutil detach -force ${device}
+  else
+    echo "Ramfs is not currently mounted at ${TIER_PATH}"
   fi
-  # Remove the "/Volumes/" part so we can get the name of the volume.
-  diskutil erasevolume HFS+ ${TIER_PATH/#\/Volumes\//} $(hdiutil attach -nomount ram://${num_sectors})
 }
 
 function mount_ramfs_local() {
+  if [[ $(uname -a) == Darwin* ]]; then
+    # Assuming Mac OS X
+    umount_ramfs_mac
+    mount_ramfs_mac
+  else
+    # Assuming Linux
+    check_space_linux
+    umount_ramfs_linux
+    mount_ramfs_linux
+  fi
+}
+
+function umount_ramfs_local() {
+  if [[ $(uname -a) == Darwin* ]]; then
+    umount_ramfs_mac
+  else
+    umount_ramfs_linux
+  fi
+}
+
+function run_local() {
   init_env
 
   if [[ ${TIER_ALIAS} != "MEM" ]]; then
     # the top tier is not MEM, skip
     exit 1
   fi
 
-  if [[ $(uname -a) == Darwin* ]]; then
-    # Assuming Mac OS X
-    mount_ramfs_mac
-  else
-    # Assuming Linux
-    mount_ramfs_linux $1
-  fi
+  mount_type=$1
+  case "$mount_type" in
+    Mount)
+      USE_SUDO=false
+      mount_ramfs_local
+      ;;
+    SudoMount)
+      USE_SUDO=true
+      mount_ramfs_local
+      ;;
+    Umount)
+      USE_SUDO=false
+      umount_ramfs_local
+      ;;
+    SudoUmount)
+      USE_SUDO=true
+      umount_ramfs_local
+      ;;
+    *)
+      echo -e ${USAGE} >&2
+      exit 1
+  esac
 }
 
 function main {
   case "$1" in
-    Mount|SudoMount)
+    Mount|SudoMount|Umount|SudoUmount)
```

Comment:
The new options don't seem to be used?

Reply:
they are used later in `run_local`

Reply:
We are expecting users to directly call `alluxio-mount`?

Reply:
Yes, is that different from how you view `alluxio-mount`?